doc/source/serve/advanced-guides/advanced-autoscaling.md

```diff
@@ -669,6 +669,100 @@ In your policy, access custom metrics via:
+
+### Enable external scaler
+
+Before using the external scaling webhook, enable it in your application configuration by setting `external_scaler_enabled: true`:
+
+```{literalinclude} ../doc_code/external_scaler_config.yaml
+---
+start-after: __external_scaler_config_begin__
+end-before: __external_scaler_config_end__
+emphasize-lines: 5
+language: yaml
+---
+```
+
+:::{warning}
+External scaling and built-in autoscaling are mutually exclusive. You can't use both for the same application. If you set `external_scaler_enabled: true`, you **must not** configure `autoscaling_config` on any deployment in that application. Attempting to use both results in an error.
+:::
+
+### API endpoint
+
+The external scaling webhook requires authentication using a bearer token. You can obtain this token from the Ray dashboard UI (typically at `http://localhost:8265`) in the Serve section.
+
+Scale a deployment by sending a POST request with the target number of replicas:
+
+```bash
+curl -X POST http://localhost:8000/api/v1/applications/{application_name}/deployments/{deployment_name}/scale \
+  -H "Authorization: Bearer <your_token>" \
+  -H "Content-Type: application/json" \
+  -d '{"target_num_replicas": 5}'
+```
+
+Replace `{application_name}` and `{deployment_name}` with your application and deployment names, and `<your_token>` with the authentication token from the Ray dashboard.
+
+The request body must conform to the [`ScaleDeploymentRequest`](https://docs.ray.io/en/latest/serve/api/doc/ray.serve.schema.ScaleDeploymentRequest.html) schema. The `target_num_replicas` field (integer, required) specifies the target number of replicas for the deployment and must be a non-negative integer.
+
+### Example - Predictive scaling
+
+Implement predictive scaling based on historical patterns or forecasts. For instance, you can preemptively scale up before anticipated traffic spikes:
+
+```python
+import requests
+from datetime import datetime
+
+def predictive_scale(
+    application_name: str,
+    deployment_name: str,
+    auth_token: str,
+    serve_endpoint: str = "http://localhost:8000"
+) -> bool:
+    """Scale based on time of day and historical patterns."""
+    hour = datetime.now().hour
+    
+    # Define scaling profile based on historical traffic patterns
+    if 9 <= hour < 17:  # Business hours
+        target_replicas = 10
+    elif 17 <= hour < 22:  # Evening peak
+        target_replicas = 15
+    else:  # Off-peak hours
+        target_replicas = 3
+    
+    url = (
+        f"{serve_endpoint}/api/v1/applications/{application_name}"
+        f"/deployments/{deployment_name}/scale"
+    )
+    
+    headers = {
+        "Authorization": f"Bearer {auth_token}",
+        "Content-Type": "application/json"
+    }
+    
+    response = requests.post(
+        url,
+        headers=headers,
+        json={"target_num_replicas": target_replicas},
+        timeout=10
+    )
+    
+    return response.status_code == 200
+```
+
```

Comment:
We should callout some important behaviors here
1. is the call to external scaler API idempotent?
2. what happens when serve deploy is called after external scaler API changes number of replicas?
3. how can this external scaling control loop know about current number of replicas. Best to include this as part of the existing code example.
4. what is the initail number of replicas when the application is deployed for the first time?